lib/pbench/test/unit/server/test_user_management_cli.py

```diff
@@ -187,40 +188,31 @@ def test_valid_user_list(client, server_config, pytestconfig):
     @staticmethod
     @responses.activate
     @pytest.mark.parametrize(
-        "email, role, firstname, lastname, username",
-        [("new_test@domain.com", "ADMIN", "newfirst", "newlast", "newuser")],
+        "switch, value",
+        [
+            (USER_SWITCH, "new_test"),
+            (EMAIL_SWITCH, "new_test@domain.com"),
+            (ROLE_SWITCH, "ADMIN"),
+            (FIRST_NAME_SWITCH, "newfirst"),
+            (LAST_NAME_SWITCH, "newlast"),
+            (LAST_NAME_SWITCH, "newuser"),
+        ],
     )
-    def test_valid_user_update(
-        client, server_config, pytestconfig, email, role, firstname, lastname, username
-    ):
+    def test_valid_user_update(client, server_config, pytestconfig, switch, value):
         with client:
             runner = CliRunner(mix_stderr=False)
-            result = TestUserManagement.register_user(server_config, pytestconfig)
-            assert result.exit_code == 0, result.stderr
+            result = TestUserManagement.register_user()
 
             result = runner.invoke(
-                user_update,
-                args=[
-                    TestUserManagement.USER_TEXT,
-                    TestUserManagement.USER_SWITCH,
-                    username,
-                    TestUserManagement.EMAIL_SWITCH,
-                    email,
-                    TestUserManagement.FIRST_NAME_SWITCH,
-                    firstname,
-                    TestUserManagement.LAST_NAME_SWITCH,
-                    lastname,
-                    TestUserManagement.ROLE_SWITCH,
-                    role,
-                ],
+                user_update, args=[TestUserManagement.USER_TEXT, switch, value],
             )
             assert result.exit_code == 0
             assert result.stdout == "User test_user updated\n"
             assert not result.stderr_bytes
 
             # Update with invalid role for the user
             result = runner.invoke(
-                user_update, args=[username, TestUserManagement.ROLE_SWITCH, "ADMN",],
+                user_update, args=["newuser", TestUserManagement.ROLE_SWITCH, "ADMN"],
             )
             assert result.exit_code == 2
```

Comment:
Not to quibble, but you're performing this test in exactly the same way six times (due to the parameterization), and it doesn't really fit under the heading of "valid user update".  Perhaps it should be moved to a new test case.  (I was going to suggest moving it to the next test case (`test_invalid_user_update()`), but that's a test of an invalid _user_, not an invalid _update_.  :-D )

Reply:
Yeah you are right I am performing the same test quite a few times like in another test case specifically for this use case `test_user_creation_with_invalid_role` yeah and this test doesn't belong to "valid user update". I guess I can just remove it

Reply:
Actually, I don't think you should remove it:  I think you test invalid role with _creation_ but not with **update**.  So, I think you should keep the scenario, but move it to its own test.  We don't want it where it is, because that test is parameterized; and, I didn't see another test which made sense for it; so, you should create a new test for it.